Vinyl/Turntable.swift

```diff
@@ -32,31 +34,55 @@ public final class Turntable: NSURLSession {
-        
-        self.init(configuration: turntableConfiguration, delegateQueue: delegateQueue)
+    public convenience init(vinyl: Vinyl, turntableConfiguration: TurntableConfiguration = TurntableConfiguration(), delegateQueue: NSOperationQueue? = nil, urlSession: NSURLSession? = nil) {
+        self.init(configuration: turntableConfiguration, delegateQueue: delegateQueue, urlSession: urlSession)
         player = Turntable.createPlayer(vinyl, configuration: turntableConfiguration)
     }
     
-    public convenience init(cassetteName: String, bundle: NSBundle = testingBundle(), turntableConfiguration: TurntableConfiguration = TurntableConfiguration(), delegateQueue: NSOperationQueue? = nil) {
-        
+    public convenience init(cassetteName: String, bundle: NSBundle = testingBundle(), turntableConfiguration: TurntableConfiguration = TurntableConfiguration(), delegateQueue: NSOperationQueue? = nil, urlSession: NSURLSession? = nil) {
         let vinyl = Vinyl(plastic: Turntable.createCassettePlastic(cassetteName, bundle: bundle))
-        self.init(vinyl: vinyl, turntableConfiguration: turntableConfiguration, delegateQueue: delegateQueue)
+        self.init(vinyl: vinyl, turntableConfiguration: turntableConfiguration, delegateQueue: delegateQueue, urlSession: urlSession)
+    }
+    
+    public convenience init(vinylName: String, bundle: NSBundle = testingBundle(), turntableConfiguration: TurntableConfiguration = TurntableConfiguration(), delegateQueue: NSOperationQueue? = nil, urlSession: NSURLSession? = nil) {
+        let plastic = Turntable.createVinylPlastic(vinylName, bundle: bundle, recordingMode: turntableConfiguration.recordingMode)
+        let vinyl = Vinyl(plastic: plastic ?? [])
+        self.init(vinyl: vinyl, turntableConfiguration: turntableConfiguration, delegateQueue: delegateQueue, urlSession: urlSession)
+        
+        switch turntableConfiguration.recordingMode {
+        case .MissingVinyl where plastic == nil, .MissingTracks:
+            recorder = Recorder(wax: Wax(vinyl: vinyl), recordingPath: recordingPath(fromConfiguration: turntableConfiguration, vinylName: vinylName, bundle: bundle))
+        default:
+            recorder = nil
+            recordingSession = nil
+        }
+    }
+    
+    deinit {
+        stopRecording()
     }
     
-    public convenience init(vinylName: String, bundle: NSBundle = testingBundle(), turntableConfiguration: TurntableConfiguration = TurntableConfiguration(), delegateQueue: NSOperationQueue? = nil) {
+    public func stopRecording() {
+        guard let _ = recordingSession else {
+            return
+        }
         
-        let plastic = Turntable.createVinylPlastic(vinylName, bundle: bundle)
-        self.init(vinyl: Vinyl(plastic: plastic), turntableConfiguration: turntableConfiguration, delegateQueue: delegateQueue)
+        recordingSession = nil
+        persistRecording()
     }
```

Comment:
I don't particularly like this approach. For two reasons:

1. The `guard let _ =` which kinda signals something is wrong.
2. This is the only method that calls `persistRecording`, it would be preferable to move that logic here:

```swift
public func stopRecording() {
        guard let recorder = recorder else {
            return
        }

        do {
           try recorder.persist()
        }
        catch {
            fatalError("💣 we couldn't save the recording.")
        }
}
```

Btw why are you nilling out `recordingSession`?